django/db/models/sql/compiler.py

```diff
@@ -532,15 +532,12 @@ def get_combinator_sql(self, combinator, all):
                         "ORDER BY not allowed in subqueries of compound statements."
                     )
         elif self.query.is_sliced and combinator == "union":
-            limit = (self.query.low_mark, self.query.high_mark)
             for compiler in compilers:
                 # A sliced union cannot have its parts elided as some of them
                 # might be sliced as well and in the event where only a single
                 # part produces a non-empty resultset it might be impossible to
                 # generate valid SQL.
                 compiler.elide_empty = False
-                if not compiler.query.is_sliced:
-                    compiler.query.set_limits(*limit)
```

Comment:
Unfortunately, it causes `queries.test_qs_combinators.QuerySetSetOperationTests.test_union_with_select_related_and_first` crash on Oracle. It seems that Oracle requires `ORDER BY` and `LIMIT/OFFSET` in all compound queries when they are wrapped in a subquery. As far as I'm aware we can amend `test_union_with_select_related_and_first` as ordering is not crucial there:
```diff
diff --git a/tests/queries/test_qs_combinators.py b/tests/queries/test_qs_combinators.py
index 6358e537ae..1ed5b46241 100644
--- a/tests/queries/test_qs_combinators.py
+++ b/tests/queries/test_qs_combinators.py
@@ -314,15 +314,14 @@ class QuerySetSetOperationTests(TestCase):
         qs2 = base_qs.filter(name="a2")
         self.assertSequenceEqual(qs1.union(qs2).order_by("pk"), [a1, a2])
 
-    @skipUnlessDBFeature("supports_slicing_ordering_in_compound")
     def test_union_with_select_related_and_first(self):
         e1 = ExtraInfo.objects.create(value=7, info="e1")
         a1 = Author.objects.create(name="a1", num=1, extra=e1)
         Author.objects.create(name="a2", num=3, extra=e1)
-        base_qs = Author.objects.select_related("extra")
+        base_qs = Author.objects.select_related("extra").order_by()
         qs1 = base_qs.filter(name="a1")
         qs2 = base_qs.filter(name="a2")
-        self.assertEqual(qs1.union(qs2).first(), a1)
+        self.assertEqual(qs1.union(qs2).order_by("name").first(), a1)
 
     def test_union_with_first(self):
         e1 = ExtraInfo.objects.create(value=7, info="e1")
```